CMakeLists.txt

```diff
@@ -621,3 +621,35 @@ if (SPIRV_CROSS_CLI)
 		endif()
 	endif()
 endif()
+
+if (NOT SPIRV_CROSS_SKIP_INSTALL)
+	file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/SPIRV-CrossConfig.cmake.in" [=[
+	    @PACKAGE_INIT@
+	    include("${CMAKE_CURRENT_LIST_DIR}/SPIRV-CrossTargets.cmake")
+	]=])
+
+	configure_package_config_file(
```

Comment:
All of this is new to me. Is there any place I can read up on CMake best practices regarding this code?

Reply:
I do not have a specific tutorial but I can share a few links and tips.

Let's start with some CMake documentation link:
- [`find_package`](https://cmake.org/cmake/help/latest/command/find_package.html): in particular the [Search Modes](https://cmake.org/cmake/help/latest/command/find_package.html#search-modes) section describes what the "Config Mode" is.
- [`CMakePackageConfigHelpers`](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html): this is a CMake utility module to assist you in writing CMake Config package. You do not have to do it and write one manually, but I have found that it is usually enough, at least to start with a first CMake Config package.

In the existing code [`install(TARGETS <target>... [...])`](https://cmake.org/cmake/help/latest/command/install.html#targets) with its `EXPORT` argument and [`install(EXPORT <export-name> [...])`](https://cmake.org/cmake/help/latest/command/install.html#export) to generate this CMake Config package instead.

I have found that it is better to use the `EXPORT` argument to define a CMake target export set like it was designed for and includes it in the generated CMake Config package file like it is done in this PR.
For example, it gives you flexibility to specify indirect dependencies using [`CMakeFindDependencyMacro`](https://cmake.org/cmake/help/latest/module/CMakeFindDependencyMacro.html) or expose package variable for user to consume.

Also, `write_basic_package_version_file` helps you to create a CMake Config Version package file which helps to resolve version compatibility when using [`find_package`](https://cmake.org/cmake/help/latest/command/find_package.html) with its `VERSION` argument.

I hope this helps !

Last but not least, here are two more maybe useful links:
- I submitted similar changes on [`KhronosGroup/glslang`](https://github.com/KhronosGroup/glslang) in KhronosGroup/glslang#2989
- I often do personal contribution on [`microsoft/vcpkg`](https://github.com/microsoft/vcpkg) which is a package manager for C++. Having a CMake Config package does really helps to have a consistent experience and to consume C++ package. Based on changes from this PR I would like to add a `vcpkg` package which would help to consume `SPIRV-Cross` for any other projects using it.